app-backend/db/src/main/scala/AnnotationDao.scala

```diff
@@ -364,4 +364,144 @@ object AnnotationDao extends Dao[Annotation] {
       )
     }
   }
+
+  def listDetectionLayerAnnotationsByTaskStatus(
+      projectId: UUID,
+      layerId: UUID,
+      taskStatuses: List[String]
+  ): ConnectionIO[List[Annotation]] = {
+    val taskStatusF: List[Option[Fragment]] = taskStatuses.map(taskStatus => {
+      Some(fr"tasks.status = ${taskStatus}::task_status")
+    })
+    (fr"""
+      SELECT 
+        annotations.id,
+        annotations.project_id,
+        annotations.created_at,
+        annotations.created_by,
+        annotations.modified_at,
+        annotations.owner,
+        project_labels.name,
+        annotations.description,
+        annotations.machine_generated,
+        annotations.confidence,
+        annotations.quality,
+        annotations.geometry,
+        annotations.annotation_group,
+        annotations.labeled_by,
+        annotations.verified_by,
+        annotations.project_layer_id,
+        annotations.task_id
+      FROM annotations
+      JOIN tasks
+      ON ST_Intersects(tasks.geometry, annotations.geometry)
```

Comment:
I would be interested in seeing the query planner here - though it could change drastically based on the size of data (likely too small in development to be too useful).

My concern is that the `JOIN` on `annotations` and `tasks` happens before the filter of tasks and annotations by `project_id` and `project_layer_id` -- which would be a pretty big join. I would suggest moving those filters to the join condition so that it would look something like:

```sql
JOIN tasks
ON ST_Intersects(tasks.geometry, annotations.geometry)
    AND tasks.project_id = ${projectId}
    AND tasks.project_layer_id = ${layerId}
    AND annotations.project_id = ${projectId}
    AND annotations.project_layer_id = ${layerId}
    AND (""" ++ Fragments.orOpt(taskStatusF: _*) ++ fr")")
```

The query planner might already do that; however, I've seen issues with its planning for spatial filters/joins. It is also possible and valid that we leave as-is for a future optimization.

Reply:
I agree that this could be a slow query. Since we already added a `task_id` field to the `annotations` table, @alkamin and I discussed, and we feel that we can populate it for all legacy annotations spatially on staging and production, so this export SQL won't need to care about spatial joins anymore. What do you think?

Reply:
I don't think it's 100 percent necessary if this works; however, it would be good to remove an extra what will be an underused code path. Do we need an issue for the migration of legacy annotations?